_posts/2017-10-31-ciw-paper.md

```diff
@@ -0,0 +1,86 @@
+reproducibility and best practice, and propose a Python library for discrete
+event simulations of queueing networks,
+[Ciw](http://ciw.readthedocs.io/en/latest/), that is designed with these in
+mind.
+
++ [**Ciw: An open source discrete event simulation library**](https://arxiv.org/abs/1710.03561)
+
+A [conference paper](https://pdfs.semanticscholar.org/2bfc/94644b45d73d9f63f0402d8c3191d0e0e154.pdf)
+by Richard A. Kilgore lists three properties as minimum requirements for
+reproducibility in simulations:
+
++ Readability
++ Modularity
++ Extendibility
+
+We add to that [***best software practices***](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745).
+
+### The Problem
+
+The current landscape of discrete event simulation practices is a large reliance
+on commercial stand alone software, which are often GUI-heavy.
+There's also a small proportion of simulation practitioners who work in Excel
+(small, but significant enough to be included in most simulation textbooks).
+
+Let's elaborate:
+
++ Commercial software have high costs. This limits model sharing to those who
+can pay.
++ Closed source software restricts access to source code, inhibiting model
+understanding and flexibility.
+
+*These properties directly go against the concept of open science.*
+
++ Stand alone packages tend not to be modular or extendible, hindering model
+re-usability.
++ Model testing is difficult, and the use of GUIs may encourage bad model
+validation and verification.
+* Binary model files make version control troublesome.
+* Models hidden behind GUIs are not readable, crucial parameters and behaviours
+are hidden behind a hierarchy of menus.
+
+### A Solution?
+
+One solution to this that we propose is the [Ciw](http://ciw.readthedocs.io/en/latest/)
+library, used in a Python ecosystem.
+
+All three properties of reproducible simulation modelling (Readability,
+Modularity, and Extendibility), along with best practices, can apply to well
+written Python, and the use of Ciw:
+
++ Open source so free to download, use, modify, and contribute.
++ All of Ciw's source code is available, modifiable, and extendible. It is also
+fully tested.
++ A simulation model is a script, so is testable, shareable, and version
+controllable.
++ We believe Ciw has clear, readable syntax.
++ Seeds of random number streams are clearly set, and data collection is
+transparent, which means everything can be saved.
++ Python is renowned in the scientific community, as it has readable syntax and
+an abundant collection of other scientific libraries.
++ Conducting simulations in this ecosystem would then allow seamless integration
+with other methodologies, such as data analysis, machine learning, and other
+algorithms.
++ Object-orientation, which Python is known for, and on which Ciw is built, is
+very agreeable with discrete event simulation. This also forces modularity, and
+encourages extendibility.
```

Comment:
Feel like you could have more here.

Perhaps give a brief summary of the paper (that you also compare to other frameworks, perhaps list them)? Or at least finish with some sort of summary.